src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -163,6 +176,55 @@ protected override void ExecuteCore()
             ReadProjectFileDependencies();
             RaiseLockFileTargets();
             GetPackageAndFileDefinitions();
+
+            FixUpNetStandardLibraryFacadeAssemblies();
+        }
+
+        /// <remarks>
+        /// The NETStandard.Library package (starting from version 2.0) adds references
+        /// to most assemblies via a .targets file rather than going through the normal
+        /// process to resolve assemblies from packages. We need to find these and
+        /// create the appropriate file definitions and dependencies to associate them
+        /// with the package.
+        /// Here we find the "facade" references from the .NETStandard.Library package,
+        /// update the existing FileDefinition, and create a FileDependency between the
+        /// assembly and the package.
+        /// </remarks>
+        private void FixUpNetStandardLibraryFacadeAssemblies()
+        {
+            var facades = References.Where(item => item.GetBooleanMetadata("Facade") == true
+                                                   && item.GetMetadata("NuGetPackageId") == "NETStandard.Library"
+                                                   && item.GetMetadata("NuGetPackageVersion") != null);
+
+            foreach (var facade in facades)
+            {
+                var packageName = facade.GetMetadata("NuGetPackageId");
+                var packageVersion = facade.GetMetadata("NuGetPackageVersion");
+                var facadePath = Path.GetFullPath(facade.ItemSpec);
+
+                var fileDefinition = _fileDefinitions
+                    .Where(def => def.GetMetadata(MetadataKeys.PackageName) == packageName
+                                  && def.GetMetadata(MetadataKeys.PackageVersion) == packageVersion
+                                  && def.GetMetadata(MetadataKeys.ResolvedPath) == facadePath)
+                    .FirstOrDefault();
```

Comment:
>.FirstOrDefault(); [](start = 20, length = 18)

The first two conditions of the "Where", as well as packageName and packageVersion, can be lifted out of the for-loop, so you are not recomputing this information for each facade (since presumable they are all part of one package). 
Also the string comparisons should be Ordinal-Ignore-Case